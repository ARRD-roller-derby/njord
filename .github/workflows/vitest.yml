name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30