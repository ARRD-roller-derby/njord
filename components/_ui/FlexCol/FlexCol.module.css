.container {
  display: flex;
  flex-direction: column;
  gap: 10px;
  justify-content: center;
}

.container[halign='true']{
  align-items: center;
}