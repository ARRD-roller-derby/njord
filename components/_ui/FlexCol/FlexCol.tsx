import classes from './FlexCol.module.css';

interface props {
  readonly children: string|JSX.Element|Array<JSX.Element>
  readonly halign?:boolean
}

export default function FlexCol({children,halign}:props){

  return <div className={classes.container} data-halign={halign}>{children}</div>
}