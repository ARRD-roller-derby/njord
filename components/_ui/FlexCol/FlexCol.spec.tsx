import { afterEach, describe, expect, it } from 'vitest'
import { render, cleanup } from '@testing-library/react'
import FlexCol from './FlexCol'

describe('<FlexCol />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(
      <FlexCol>
        <p>test</p>
        <p>test 2</p>
      </FlexCol>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
