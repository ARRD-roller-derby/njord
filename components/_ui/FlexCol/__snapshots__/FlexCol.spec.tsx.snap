// Vitest Snapshot v1

exports[`<FlexCol /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_1i13q_1"
  >
    <p>
      test
    </p>
    <p>
      test 2
    </p>
  </div>
</DocumentFragment>
`;
